### PR TITLE
Add ability to pass in RRule string to `--rrule` option in `prefect set-schedule` command

### PR DESCRIPTION
**Pull request by serinamarie**
### Overview
This enhancement:
1. Allows support for either a valid RRule string or dictionary (we currently only support the latter).
2. Handles each type of schedule more explicitly and checks for and returns clearer error messages sooner to the user, which builds slightly off of interval handling adjustments here: https://github.com/PrefectHQ/prefect/pull/8524. 

### Example
Currently, when using `prefect deployment set-schedule` to add or update a schedule on a deployment, you must pass in a JSON string to the `--rrule` option, like so: 

```python
prefect deployment set-schedule 'flow-name/deployment-name' \
--rrule '{"rrule": "DTSTART;TZID=America/Chicago:20221027T124000\nRRULE:FREQ=DAILY", \
"timezone": "America/Chicago"}'
```
One user describes their pain point around this here: https://github.com/PrefectHQ/prefect/issues/7406
It can be intuitive for users to desire to pass in a valid RRule to the `--rrule` option, like so:
```python
prefect deployment set-schedule 'flow-name/deployment-name' \
--rrule 'DTSTART:20220910T110000\nRRULE:FREQ=HOURLY;BYDAY=MO,TU,WE,TH,FR,SA' \
--timezone "America/Chicago"
```
```python
prefect deployment set-schedule 'flow-name/deployment-name' \
--rrule 'FREQ=HOURLY'
```

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: src/prefect/cli/deployment.py [hunk 1]
```diff
@@ -267,36 +267,53 @@ async def set_schedule(
...
+        )
 
-    updated_schedule = {k: v for k, v in updated_schedules[0].items() if v is not None}
+    if rrule_string is not None:
+        # a timezone in the `rrule_string` gets ignored by the RRuleSchedule constructor
+        if "TZID" in rrule_string and not timezone:
+            exit_with_error(
+                "You can provide a timezone by providing a dict with a `timezone` key to the --rrule option. E.g. {'rrule': 'FREQ=MINUTELY;INTERVAL=5', 'timezone': 'America/New_York'}."
+                "\nAlternatively, you can provide a timezone by passing in a --timezone argument."
+            )
```
Comment (serinamarie) [inline src/prefect/cli/deployment.py:309]: Wondering about this 🤔

File: src/prefect/cli/deployment.py [hunk 2]
```diff
@@ -267,36 +267,53 @@ async def set_schedule(
...
 
-    updated_schedules = list(
-        filter(
-            updated_schedule_check, (interval_schedule, cron_schedule, rrule_schedule)
+    if interval is not None:
+        if interval_anchor:
+            try:
+                pendulum.parse(interval_anchor)
+            except ValueError:
+                exit_with_error("The anchor date must be a valid date string.")
```
Comment (madkinsz) [inline src/prefect/cli/deployment.py:283]: Does this need to be updated now that `main` has these changes?

Comment (serinamarie) [reply src/prefect/cli/deployment.py:283]: Yes!

**Comment by rpeden** [comment]
I pulled and QA'ed the changes, and things worked as expected. Plain RRule strings work, as does the JSON format that was previously required.

**Review by rpeden — APPROVED** [review]
As mentioned, looks good and everything worked when I QA'ed.

**Pull request merged**